os/drivers/serial/serial.c

```diff
@@ -109,6 +111,18 @@ static int uart_poll(FAR struct file *filep, FAR struct pollfd *fds, bool setup)
  * Private Variables
  ************************************************************************************/
 
+#ifdef CONFIG_DYNAMIC_LOGS_STORING
+static sq_queue_t log_net_chunks;
+static bool log_net_chunks_initialized = false;
+static size_t log_net_size = 0;
+
+struct log_net_chunk_s {
+	struct log_net_chunk_s *flink;
+	int curbytes;
```

Comment:
I don't know why this is necessary. Could you let me know why current location of buffer is necessary for each? This is valid only for current chunk, but other old chunk is always max (CONFIG_LOG_CHUNK_SIZE].
Because we can allocate max number of chunk, 256, if we can remove this, then we can save much memory.

Reply:
You are correct. Its only required for the current chunk that is being filled. I will remove it from the struck and make it a global variable as it is used only for the last chunk.